### PR TITLE
chore: improve deployment in dev container

### PR DESCRIPTION
**Pull request by cre8ivejp**
I'm improving the steps to deploy Bucketeer and run E2E tests into a few steps only.
E2E tests passed running in the dev container.

### Things done

- Removed unused `service-token-secret.yaml`
- Updated the dev container to use the secret instead.
- Changed the local domain from `bucketeer.org` to `bucketeer.io`
- Deleted unnecessary rules from Makefile
- Pushed the TLS, OAUTH, and the service token for DEV container
- Updated the DEVELOPMENT.md

---

This pull request includes changes to the `DEVELOPMENT.md`, `Makefile`, and various YAML files in the `manifests/bucketeer/charts` directory. The changes mainly focus on simplifying the setup and deployment process for the Bucketeer project, improving the grammar and clarity of the instructions, and updating the domain from `bucketeer.org` to `bucketeer.io`.

Key Changes:

Grammar and Clarity Improvements:
* [`DEVELOPMENT.md`](diffhunk://#diff-fc2a310fcfedfefb0046def697f932def80cbb1e78c689bc0af3c8eab3e33eceL1-R110): Improved the grammar and clarity of the instructions for setting up the development environment, deploying Bucketeer, and running E2E tests.

Domain Update:
* `DEVELOPMENT.md`, `Makefile`, various YAML files: Updated the domain from `bucketeer.org` to `bucketeer.io`. [[1]](diffhunk://#diff-fc2a310fcfedfefb0046def697f932def80cbb1e78c689bc0af3c8eab3e33eceL1-R110) [[2]](diffhunk://#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52L286-R287) [[3]](diffhunk://#diff-72e77939e3c1ff66fcda157b964a2c65ac5b38801112c8d03f87a247a5aee2e8L26-R35) [[4]](diffhunk://#diff-4c42fb2cdeb0ff66a5f05236d2ad2b6464824fd5a122618ededbf6ff460c2bf3L59-R58) [[5]](diffhunk://#diff-e22de46129c31ba83d35848cbda70471be471b32ce7f95022d4d5fb971bbf086L104)

Simplification of Setup and Deployment Process:
* [`Makefile`](diffhunk://#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52L308-L327): Removed unnecessary commands and simplified the `deploy-bucketeer` command to improve the setup and deployment process. [[1]](diffhunk://#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52L308-L327) [[2]](diffhunk://#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52L361-R355) [[3]](diffhunk://#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52L339) [[4]](diffhunk://#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52L361-R355)
* Various YAML files: Removed the service token secrets and updated the `values.dev.yaml` files to point to the correct service token secret. [[1]](diffhunk://#diff-55103ebf058a180f3c5bafce127e1f3c2b99a284e1cd8546b1a89d0bbc8d56afL1-L15) [[2]](diffhunk://#diff-4c42fb2cdeb0ff66a5f05236d2ad2b6464824fd5a122618ededbf6ff460c2bf3L48-R48) [[3]](diffhunk://#diff-e22de46129c31ba83d35848cbda70471be471b32ce7f95022d4d5fb971bbf086L53-R53) [[4]](diffhunk://#diff-3d8bde9caf98d171e121808895b33c07846db0ad3e3c03b14aaa5b707f2dc362L33-R33)

Other Changes:
* [`Makefile`](diffhunk://#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52L361-R355): Updated the `delete-all-services-from-minikube` command to ignore services that are not found.
* Various `values.yaml` and `values.dev.yaml` files: Removed unused `token` fields. [[1]](diffhunk://#diff-8aa57c31b5f237120424a1cc3a25c800578f13c9468532c6522a251eb6bf87a5L91) [[2]](diffhunk://#diff-da142715fb447fa09cd1b8854c0f9caa732a8d7d0555e25ae2bd0d8cb52d4b4dL82) [[3]](diffhunk://#diff-c1ee9113ab7bc77fd7654dab7356dcbcd80348631b519ce03db52cb725b32663L73)

### CODE REVIEW

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -1,166 +1,112 @@
-## Setup development environment
+# Set up the DEV Container
 
-It's recommended to use Github dev container to setup the development environment.
+It's recommended that the GitHub dev container be used to set up the development environment.
```
Comment (kentakozuka) [inline DEVELOPMENT.md:3]:
```suggestion
It's recommended that Development Container (Dev Container) be used to set up the development environment.
```

**Review by kentakozuka — DISMISSED** [review]
Thank you!  left a nit.

Also, it would be nice to have default values of make commands such as the following variables;
```
WEB_GATEWAY_URL=web-gateway.bucketeer.io \
GATEWAY_URL=api-gateway.bucketeer.io \
WEB_GATEWAY_CERT_PATH=/workspaces/bucketeer/tools/dev/cert/tls.crt \
GATEWAY_CERT_PATH=/workspaces/bucketeer/tools/dev/cert/tls.crt \
SERVICE_TOKEN_PATH=/workspaces/bucketeer/tools/dev/cert/service-token \
API_KEY_PATH=/workspaces/bucketeer/tools/dev/cert/api_key_client \
API_KEY_ROLE=SDK_CLIENT \
ENVIRONMENT_NAMESPACE=e2e \
```

File: .gitignore [hunk 2]
```diff
@@ -9,4 +9,5 @@ node_modules
 telepresence.log
 venv
 *.swp
-tools/dev/cert/*
\ No newline at end of file
+tools/dev/cert/api_key_client
+tools/dev/cert/api_key_server
```
Comment (Ubisoft-potato) [inline .gitignore:13]: Api key shouldn't be pushed to repo, because the api key is store in Dev container's database.

File: Makefile [hunk 3]
```diff
@@ -358,25 +347,20 @@ deploy-service-to-minikube:
...
-	EMAIL=$(EMAIL) \
-	OAUTH_KEY_PATH=/workspaces/bucketeer/tools/dev/cert/oauth-private.pem \
-	SERVICE_TOKEN_PATH=/workspaces/bucketeer/tools/dev/cert/service-token \
-	make generate-service-token
+# Bucketeer deployment
+deploy-bucketeer: delete-all-services-from-minikube
+	make -C tools/dev service-cert-secret
+	make -C tools/dev service-token-secret
+	make -C tools/dev oauth-key-secret
+	GITHUB_TOKEN=$(GITHUB_TOKEN) make -C tools/dev generate-github-token
```
Comment (Ubisoft-potato) [inline Makefile:361]: We can also remove the github token after merge db migration PR.

This github token is used by old database migration: https://github.com/bucketeer-io/bucketeer/blob/main/pkg/backend/cmd/server/server.go#L302.

Comment (cre8ivejp) [reply Makefile:361]: I will remove it later.

**Review by Ubisoft-potato — APPROVED** [review]
LGTM!

**Pull request merged**